filecoin-proofs/parameters.json

```diff
@@ -1,82 +1,122 @@
-  },
-  "v24-proof-of-spacetime-election-PoseidonHasher-0b0b9781bcb153efbb3cab4be3a792c4f555d4ab6f8dd62b27e1dcad08a34f22.vk": {
-    "cid": "QmVXv4Q1T3FbiY5AUgWER11Lsrby9aUVJy2mgWDWrndFbq",
-    "digest": "223dd87c6161c45daf448ca9eda28298",
-    "sector_size": 34359738368
-  },
-  "v24-proof-of-spacetime-election-PoseidonHasher-0b499a953f1a9dcab420b3ba1e6b1f3952dc7f17cf67ed10406ae9a43e2b8ec5.params": {
-    "cid": "Qmea7VsrYnkrpdMnutkGKppX5finoDwCA2fP5Zg5bDuBQw",
-    "digest": "3de5b8738a2cd933c214fa2023e30909",
-    "sector_size": 8388608
+  "v25-proof-of-spacetime-fallback-MerkleTree<PoseidonHasher, 8, 0, 0>-0170db1f394b35d995252228ee359194b13199d259380541dc529fb0099096b0.params": {
+    "cid": "QmNUKXCEcjMRh8ayFG2X9RYUuc2SK5XRVsSVTqJmNWAgSp",
+    "digest": "fe10d43b607dd6687f30428476076ebb",
+    "sector_size": 2048
   },
-  "v24-proof-of-spacetime-election-PoseidonHasher-0b499a953f1a9dcab420b3ba1e6b1f3952dc7f17cf67ed10406ae9a43e2b8ec5.vk": {
-    "cid": "QmavFXmf3jeacHKB6HoJH3gUqzmKnsDn5F5HSYfwPbDHRu",
-    "digest": "485b7eab4f70031fdda4eaeccfe4f26e",
-    "sector_size": 8388608
+  "v25-proof-of-spacetime-fallback-MerkleTree<PoseidonHasher, 8, 0, 0>-0170db1f394b35d995252228ee359194b13199d259380541dc529fb0099096b0.vk": {
+    "cid": "QmRyV1DvF57cSnnwUoocKbPiULoLdfnfWpVWi8BSsMN6KR",
+    "digest": "8aaca32ca9a1c6a431b99e695b443e69",
+    "sector_size": 2048
   },
-  "v24-proof-of-spacetime-election-PoseidonHasher-27a7fc680a47e4821f40cf1676fb80b9888820ef6867a71a175b4c9ae068ad3f.params": {
-    "cid": "QmQrUjB9NSMuThe1JHoHfC7u1xdoLS6WLu15waWcnZ3tQT",
-    "digest": "7e6adc7cbf73db8c95a54e3c23bea1ae",
+  "v25-proof-of-spacetime-fallback-MerkleTree<PoseidonHasher, 8, 0, 0>-0cfb4f178bbb71cf2ecfcd42accce558b27199ab4fb59cb78f2483fe21ef36d9.params": {
+    "cid": "QmTvwEyFVcjivKUX9AqZrC4mfjLSN2JJTucLJfNaWqCPmD",
+    "digest": "1cc1bf83c9e3d9b2d994ad2ec946a79f",
     "sector_size": 536870912
   },
-  "v24-proof-of-spacetime-election-PoseidonHasher-27a7fc680a47e4821f40cf1676fb80b9888820ef6867a71a175b4c9ae068ad3f.vk": {
-    "cid": "QmVPPk4fBcEero2GHsYuBoh97yhugTBWUp9yWSPPWjRWQ7",
-    "digest": "952b352d694d650e912b3b92ad63f7c9",
+  "v25-proof-of-spacetime-fallback-MerkleTree<PoseidonHasher, 8, 0, 0>-0cfb4f178bbb71cf2ecfcd42accce558b27199ab4fb59cb78f2483fe21ef36d9.vk": {
+    "cid": "QmVfgowqdh3ruAHqQ8LA6L4VdSYwam5e8VmSEtZXBoAudC",
+    "digest": "377659f83c6714703b17828f603038fc",
     "sector_size": 536870912
   },
-  "v24-proof-of-spacetime-election-PoseidonHasher-5916054ae98e28fc2f0470d1fb58eb875a6865be86f0b8c4e302d55f13217fef.params": {
-    "cid": "QmSXMF85mdGLQfAY98zVL4dUBpGPFFUPDmFzdc1NZrVFdh",
-    "digest": "a93de0f8cfb04af5d21f66ef48ee59a8",
+  "v25-proof-of-spacetime-fallback-MerkleTree<PoseidonHasher, 8, 0, 0>-3ea05428c9d11689f23529cde32fd30aabd50f7d2c93657c1d3650bca3e8ea9e.params": {
+    "cid": "QmQ2HrKCWbtWQNNQiBj3BFE8QrqMyed8P5Vw5vyyzuSMsF",
+    "digest": "2e15ec3fbff51abf66d241252fb8babd",
     "sector_size": 2048
   },
```

Comment:
This one and the one above (""v25-proof-of-spacetime-fallback-MerkleTree<PoseidonHasher, 8, 0, 0>-0170db1f394b35d995252228ee359194b13199d259380541dc529fb0099096b0.params") seem to be duplicated but for the digest part of the cache key. Is it possible that two different circuit-code versions got included in this manifest?

Reply:
I believe this is correct, there are two fallback posts per sector size, one for window and one for winning.

Reply:
There should be and there are 24 entrie in total

- 4 sector sizes
- 3 different proofs per sector size
- 2 files (`.params`, `.vk`) per proof

Reply:
Okay. They really should have distinct names, in my opinion.

Reply:
By that, I mean there should be enough metadata surfaced in their cache keys to make it clear which is which.

Reply:
they use the same circuit, so this is not easily possible

Reply:
Don't they differ in public parameters, though (they should)? The cache identifier is a function of public parameters for that reason, so it seems like it should be possible. What am I missing?